page-objects/modals/createSpaceModal.ts

```diff
@@ -17,14 +18,17 @@ export class CreateSpaceModal {
     this.sameNameErrorMessage = this.page.getByText("Space name already exists");
   }
 
+  @logTyping("Space name")
   async typeSpaceName(newSpaceName: string) {
     await this.nameInput.fill(newSpaceName);
   }
 
+  @logClickingOnElement("'Continue' button")
   async clickOnContinueButton() {
     await this.continueButton.click();
   }
 
+  @logClicking("button")
```

Comment:
Da się tutaj jakoś bardziej opisowo? Jeśli nie można wrzucić `buttonName` w logi to może jakoś dodać info, że to button związany z jakimś modalem.

Reply:
Generalnie chodziło mi o to, że do klikania mamy obecnie jakby dwa podejścia:

1. "konkretne" metody do klikania w np. dany przysick
```typescript
async clickOnContinueButton() {
  await this.continueButton.click();
}
```

2. "ogólne" metody do klikania w coś, czego nazwę podaje się przez argument
```typescirpt
async clickOnButton(buttonName: string) {
  this.button = this.modalContainer.getByRole("button", { name: buttonName });
  await this.button.click();
}
```

W pierwszym przypadku, żeby logowany tekst niósł dobre informacje i był czytelny, dobrze byłoby podać w co się klika (nazwę i typ elementu), bo nie można tego odczytać z argumentu metody (bo nie ma żadnych argumentów). Do tego więc służy dekorator `@locClickingOnElement`, który przyjmuje tekst określający w co się klika, wraz z typem (np. `'Continue' button`.

W drugim przypadku, żeby logowana wiadomość była okej, wystarczy podać typ elementu w który się klika, bo nazwę elementu można odczytać z argumentu logowanej metody (dekorator ma dostęp do argumentów metody, którą dekoruje). Do tego służy więc dekorator `@logClicking` który wymaga tylko podania typu elementu, w który się klika (wystarczy samo np. `'button'`).

*Oba te dekoratory mają jeszcze drugi opcjonalny parametr, który precyzuje, którym przyciskiem myszy kliknięto, np. `'Right'`.*

Dodatkowo w logach wyświetla się (a przynajmniej powinna xd) "ścieżka", w sensie klasa i metoda, w której wywołano loggera, wraz z datą; przykładowe logi wyglądają tak:
```bash
20.06.2024 - 09:57:32 - [LeftMenu.clickOnElement]  Clicking on 'Create Space' left menu option
20.06.2024 - 09:57:33 - [CreateSpaceModal.typeSpaceName] Typing text 'GUI TEST new space' into 'Space name' input
20.06.2024 - 09:57:33 - [CreateSpaceModal.clickOnContinueButton]  Clicking on 'Continue' button
20.06.2024 - 09:57:33 - [CreateSpaceModal.clickOnButton]  Clicking on 'Create Space' button
```

Reply:
imo spoko rozwiązanie. Zrozumiałe podejście i logi wyglądają elegancko. Do tego nie trzeba się zastanawiać czy wrzucać log przed czy po akcji.